FWCore/MessageService/plugins/MessageLogger.cc

```diff
@@ -325,6 +325,29 @@ namespace edm {
       nonModule_errorEnabled = messageDrop->errorEnabled;
     }  // ctor
 
+    void MessageLogger::setThreadContext(ModuleCallingContext const& iModContext) {
+      //need to know if we are in a global or stream context
+      auto top = iModContext.getTopModuleCallingContext();
+      assert(nullptr != top);
+      if (ParentContext::Type::kGlobal == top->type()) {
+        auto globalContext = iModContext.getGlobalContext();
+        auto tran = globalContext->transition();
+        if (tran == GlobalContext::Transition::kBeginLuminosityBlock or
+            tran == GlobalContext::Transition::kEndLuminosityBlock) {
+          establishModule(lumiInfoBegin_ + globalContext->luminosityBlockIndex(),
+                          iModContext,
+                          s_globalTransitionNames[static_cast<int>(tran)]);
+        } else {
+          establishModule(
+              runInfoBegin_ + globalContext->runIndex(), iModContext, s_globalTransitionNames[static_cast<int>(tran)]);
+        }
```

Comment:
Would this need any specific treatment for ProcessBlock?

Reply:
I don't know. I didn't look to see if David extended the message logger for that case.

Reply:
@wddgit Could you comment?

Reply:
OK. I will take a look at this one after lunch. There were a few MessageLogger changes I vaguely remember making in the PR, but not many. It might have actually been in the first PR...

I've been trying to finish up this documentation and some slides for next week's Core meeting, before I start addressing the PR comments.

Reply:
It looks like ProcessBlock doesn't work with the interface. That is because, in my opinion, the ProcessBlock transitions in the MessageLogger service are calling the wrong `establishModule` and `unestablishModule` calls of the messaging system. They should be calling the ones using the `ModuleCallingContext` but they are using the `ModuleDescription` instead.

Reply:
Possibly you are correct. There is not anything related to message logger in the current PR. The PR merged back in July 2020 has all the MessageLogger related changes. I am looking at this now. Do you want to fix this or do you want me to fix it?

Reply:
Is this something new? Or was it incorrect a year ago also? I vaguely remember testing it and it seemed to work.

Reply:
One comment. For ProcessBlock it always a global context. There are no stream transitions. I am not sure if that is relevant here yet.

Reply:
> Or was it incorrect a year ago also? 

I believe it was incorrect a year ago. I would expect the ProcessBlock transitions to work like the Run or LuminosityBlock transitions, not like the constructor or beginJob transitions.

Reply:
I am still looking and trying to remember why I did it this way. There is a unit test and at least for beginProcessBlock and endProcessBlock it seems to be working:

https://cmssdt.cern.ch/lxr/source/FWCore/MessageService/test/unit_test_outputs/u33_all.log
lines 7 and 93

There is no accessInputProcessBlock transition test.

At some level, I intuitively see why you would it expect it to be like Run and LuminosityBlock. Although there are a lot of similarities to beginJob and endJob also. The ProcessBlock transitions are always global, no stream transitions. The modules can run concurrently but otherwise no concurrency, I am not sure how that plays into it. Still looking...

Reply:
I see three differences in the establishModule/unestablishModule functions.

1. Setting runEvent. This seems irrelevant to the ProcessBlock transitions.
2. Setting streamID. Also irrelevant for the ProcessBlock transitions.
3. Different actions if previousModuleOnThread is set on unestablishModule. I'm less sure, but I think that is also irrelevant for ProcessBlock transitions. Is there a way for the previousModuleOnThread to be set for ProcessBlock transitions?

Everything else in the function overloads seems the same. Unless I am missing something, the ModuleDescription overloads are correct and all is OK.

Reply:
The `establishModule` gets a `ModuleCallingContext` which is needed if a module can get data from another module therby cause the other module to be run. The context allows one to trace that.